src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_TupleCreationExpression.cs

```diff
@@ -14,7 +10,40 @@ internal partial class LocalRewriter
         public override BoundNode VisitTupleCreationExpression(BoundTupleCreationExpression node)
         {
             var rewrittenArguments = VisitList(node.Arguments);
-            return new BoundObjectCreationExpression(node.Syntax, node.ConstructorOpt, rewrittenArguments);
+            BoundObjectCreationExpression nextCreation = ExpandTupleConstructor(node, rewrittenArguments);
+
+            return nextCreation;
+        }
+
+        /// <summary>
+        /// Converts the expression for creating a tuple instance into an expression creating a ValueTuple (if short) or nested ValueTuples (if longer).
+        ///
+        /// For instance, for a long tuple we'll generate:
+        /// creationExpression(ctor=largestCtor, args=firstArgs+(nested creationExpression for remainder, with smaller ctor and next few args))
+        /// </summary>
+        private static BoundObjectCreationExpression ExpandTupleConstructor(BoundTupleCreationExpression node, ImmutableArray<BoundExpression> rewrittenArguments)
+        {
+            var numChunks = TupleTypeSymbol.NumberOfValueTuples(rewrittenArguments.Length);
+
+            var nextCreation = default(BoundObjectCreationExpression);
+            for (int chunkIndex = numChunks - 1; chunkIndex >= 0; chunkIndex--)
+            {
+                ImmutableArray<BoundExpression> chainedArguments;
+
+                if (chunkIndex == numChunks - 1)
```

Comment:
Consider eliminating conditional logic from the loop. The part for ```chunkIndex == numChunks - 1``` should always be performed. So, let's move this out of the loop (before it, along with BoundObjectCreationExpression creation) and adjust the loop initialization accordingly. As an added benefit, we will be able to assign real value to nextCreation right at the declaration.